#### About

> Power of a statistical test

In statistics, the power of a statistical test is the probability of rejecting a null hypothesis when it is actually false (i.e., correctly detecting a real effect). In other words, it is the probability of correctly identifying a significant difference or relationship in a study.

The power of a statistical test is influenced by several factors, including the sample size, effect size, level of significance, and variability of the data.




> Example

Suppose we want to test if there is a difference in the mean height between men and women in a population. We randomly select a sample of 50 men and 50 women and measure their heights. We use a t-test to test the null hypothesis that there is no difference in the mean height between men and women against the alternative hypothesis that there is a difference. We set the significance level at 0.05.



In [5]:
import numpy as np
from scipy import stats
from statsmodels.stats.power import tt_ind_solve_power


In [2]:
male_heights = np.random.normal(175, 8, 50)
female_heights = np.random.normal(162, 7, 50)


In [3]:
# Calculate t-statistic and p-value
t_stat, p_val = stats.ttest_ind(male_heights, female_heights)

In [6]:
# Calculate power of the test
effect_size = abs(male_heights.mean() - female_heights.mean()) / np.sqrt((male_heights.var() + female_heights.var()) / 2)
power = tt_ind_solve_power(effect_size=effect_size, nobs1=50, alpha=0.05, ratio=1, alternative='two-sided')


In [7]:
print(f"t-statistic: {t_stat:.2f}, p-value: {p_val:.4f}, power: {power:.2f}")


t-statistic: 7.13, p-value: 0.0000, power: 1.00


The output shows that the t-statistic is 7.13 and the p-value is very small (less than 0.05), indicating strong evidence against the null hypothesis. The power of the test is 1.00, indicating that the test has a high probability of correctly detecting a difference in the mean height between men and women, given the sample size and effect size.